tests/functional/test_eclab_parser.py

```diff
@@ -0,0 +1,406 @@
+        "Ns changes": ("", (0, 0)),
+        "counter inc.": ("", (0, 0)),
+        "Ns": ("", (0, 0)),
+        "time/s": ("s", (5.109679870918626e001, 1.122481971643720e002)),
+        "control/mA": ("mA", (0.0000000e000, 0.0000000e000)),
+        "Ewe/V": ("V", (8.1625509e-001, 8.4291708e-001)),
+        "I/mA": ("mA", (2.7354923e-005, 3.2436059e-005)),
+        "dQ/C": ("C", (0.0000000e000, 3.2346591e-007)),
+        "(Q-Qo)/C": ("C", (0.0000000e000, 3.2346591e-007)),
+        "half cycle": ("", (0, 0)),
+        "Q charge/discharge/mA.h": (
+            "mA.h",
+            (0.000000000000000e000, 8.985164472556385e-008),
+        ),
+        "I Range": ("", (41, 41)),
+        "Ece/V": ("V", (1.0758879e000, 1.0688440e000)),
+        "Q discharge/mA.h": ("mA.h", (0.000000000000000e000, 0.000000000000000e000)),
+        "Q charge/mA.h": ("mA.h", (0.000000000000000e000, 8.985164472556385e-008)),
+        "Capacity/mA.h": ("mA.h", (0.000000000000000e000, 8.985164472556385e-008)),
+        "Efficiency/%": ("%", (0.0000000e000, 0.0000000e000)),
+        "cycle number": ("", (0.000000000000000e000, 0.000000000000000e000)),
+        "P/W": ("W", (2.2328596e-008, 2.7340908e-008)),
+        "Ewe-Ece/V": ("V", (-2.5963283e-001, -2.2592688e-001)),
+        "loop_number": ("", (0, 1)),  # added by Ixdat
+    },
+}
+
+
+@fixture(scope="module", params=tuple(TEST_DATA.items()))
+@mark.external
+def measurements_with_data(request):
+    """Load all measurements from files and connect it with the test data."""
+
+    filename = request.param[0]
+    column_data = request.param[1]
+    measurement = Measurement.read(DATA_DIR / filename, reader="biologic")
+    return measurement, column_data
+
+
+@mark.external
+def test_shape(measurements_with_data):
+    """Test the shape of the parsed data from the measurements.
+
+    The method is testing whether the amount and the names
+    of the columns are the same.
+
+    """
+    measurement = measurements_with_data[0]
+    column_names = measurements_with_data[1].keys()
+
+    # test amount of columns
+    assert len(measurement.series_list) == len(column_names)
+
+    # test names of columns
+    for name in column_names:
+        if name not in measurement.series_names:
+            assert (
+                False
+            ), f"Parsed value does not match with the value in a column: {name}"
+
+
+@mark.external
+def test_units(measurements_with_data):
+    """Test the units of the parsed data from the measurements.
+
+    The method is testing whether the units of the parsed data
+    in the columns are the same.
+
+    """
+    measurement = measurements_with_data[0]
+    column_data = measurements_with_data[1]
+
+    for column_name, unit_values in column_data.items():
+        expected_unit = unit_values[0]
+
+        assert (
+            expected_unit == measurement[column_name].unit_name
+        ), f"Parsed value does not match with the value in a column: {column_name}"
```

Comment:
If you want to expand with an error message here, I think it should be more specific, like mention that it is the unit which doesn't match

Reply:
Well, the pytest is telling us that it is the test for units that failed, but it does not say which column has the different unit value. So I wanted to make it more clear which column to look at.
I will make the assert messages more specific.